lib/date.js

```diff
@@ -98,8 +98,17 @@ module.exports = (key, options) => {
         label: label === labelKey ? v.label : label
       });
     }));
+    const conditionalTranslate = (fieldKey) => {
+      let result = req.translate(fieldKey);
+      if (result === fieldKey) {
+        result = null;
+      }
+      return result;
+    };
```

Comment:
this is a `pure` function - meaning whenever it is called with the same arguments, it will return the same value. This is the case here, but if you were to rely on an external argument not passed into the function, it would no longer be pure - an example of this would be:

```js
const key = 'field-name';

const conditionalRender = (part, translate) => {
  const translationKey = `fields.${key}.${part}`;
  let result = translate(translationKey);
  if (result === translationKey) {
    result = null;
  }
  return result;
}
```

^ This function relies on the external constant `key`, the function will return different results, if called with the same arguments but `key` is changed.

To sum up, impure functions are more bug-prone as they are dependent on external variables/factors